proto/umee/incentive/v1/genesis.proto

```diff
@@ -0,0 +1,74 @@
+import "gogoproto/gogo.proto";
+import "umee/incentive/v1/incentive.proto";
+
+// GenesisState defines the x/incentive module's genesis state.
+message GenesisState {
+  Params                    params               = 1 [(gogoproto.nullable) = false];
+  repeated IncentiveProgram programs             = 2 [(gogoproto.nullable) = false];
+  uint32                    next_program_id      = 3;
+  repeated cosmos.base.v1beta1.Coin total_locked = 4 [
+    (gogoproto.nullable)     = false,
+    (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins"
+  ];
+  repeated LockAmount        lock_amounts        = 5 [(gogoproto.nullable) = false];
+  repeated PendingReward     pending_rewards     = 6 [(gogoproto.nullable) = false];
+  repeated RewardBasis       reward_bases        = 7 [(gogoproto.nullable) = false];
+  repeated RewardAccumulator reward_accumulators = 8 [(gogoproto.nullable) = false];
+  repeated Unbonding         unbondings          = 9 [(gogoproto.nullable) = false];
+}
+
+// LockAmount tracks the amount of coins of one uToken denomination locked into a
+// given tier by a single supplier.
+message LockAmount {
+  string                   supplier = 1;
+  uint32                   tier     = 2;
+  cosmos.base.v1beta1.Coin amount   = 3 [(gogoproto.nullable) = false];
+}
+
+// PendingReward tracks the amount of rewards that a given supplier has calculated
+// but not yet claimed.
+message PendingReward {
+  string   supplier                                = 1;
+  repeated cosmos.base.v1beta1.Coin pending_reward = 2 [
+    (gogoproto.nullable)     = false,
+    (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins"
+  ];
+}
+
+// RewardBasis tracks the value of a given tier and lock denom's RewardAccumulator
+// at the last time a specific supplier calculated pending rewards for it.
+message RewardBasis {
+  string   supplier                                 = 1;
+  uint32   tier                                     = 2;
+  string   lock_denom                               = 3;
+  repeated cosmos.base.v1beta1.DecCoin reward_basis = 4 [
+    (gogoproto.nullable)     = false,
+    (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.DecCoins"
+  ];
+}
+
+// RewardAccumulator is a global reward tracking struct that indicates the amount
+// of rewards that a single unit of lock_denom would have acucmulated if it was
+// locked at a given tier since genesis.
+message RewardAccumulator {
+  uint32   tier                                     = 1;
+  string   lock_denom                               = 2;
+  repeated cosmos.base.v1beta1.DecCoin reward_basis = 4 [
+    (gogoproto.nullable)     = false,
+    (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.DecCoins"
+  ];
+}
+
+// Unbonding is a structure that tracks an in-progress token unbonding.
+message Unbonding {
+  string                   supplier = 1;
+  uint32                   tier     = 2;
+  uint64                   end      = 3;
```

Comment:
Incentive programs start and end at block heights, so I have unbondings working the same way. This will prevent bursts of activity instantly from happening (potentially out of order) after any chain halts.

Reply:
To be honest, I'm not sure what's better: in case of a 3 day halt (for example) - do we want to commit and compute farming using blocks or seconds? 
I think seconds is more logical - because the time passed and stakers didn't have access to their founds. Seconds is the "real" time.

Reply:
The motivation reduces to the same reason we have _real time_ (seconds) in staking. I don't think we should re-invent this thing.

Reply:
Switched to unix time for all